challenge-10/challenge-10.js

```diff
@@ -1,113 +1,203 @@
-"sum" agora é uma função, e, se o sinal correto não foi passado para a
-função "calculator", "sum" será false. Certifique-se de que "sum" não é
-"false", e então atribua às variáveis "number1" e "number2", dois números
-que serão os operandos da operação de soma.
-Após isso, mostre no console o resultado da operação, passando dois
-parâmetros para o método "log" de "console":
-- O primeiro será a mensagem da operação (usando a função criada acima);
-- O segundo, a função de soma, passando os dois operandos.
-- Se "sum" for "false", mostrar no console a mensagem de erro.
-*/
-// ?
-
-/*
-Repita desde o "PASSO 2" com as operações de subtração, multiplicação,
-divisão e resto. Crie variáveis com os nomes "subtraction",
-"multiplication", "division" e "mod".
-*/
-// ?
-
-/*
-Repita o PASSO 2 novamente, mas passando um operador inválido, para ver se
-a mensagem de erro será mostrada no console.
-*/
-// ?
\ No newline at end of file
+(function(){
+	/*
+	Crie uma IIFE que envolva todo esse arquivo (inclusive esse comentário),
+	e faça a indentação correta.
+	*/
+
+	/*
+	Sem alterar os códigos nos `console.log` abaixo, faça com que o retorno
+	deles seja "true", usando os Wrapper Objects como "conversores" nos valores
+	das variáveis. Analise o que está sendo impresso no console para saber como
+	resolver o problema corretamente.
+	*/
+	var five = Number('5');
+	console.log( five + ' é número?', typeof five === 'number' );
+
+	var concat = String( 10 ) + ' + ' + String( 10 );
+	console.log( '"' + concat + '" é uma string? E é igual a "1010"?', typeof concat === 'string' );
+
+	/*
+	Voltando ao exemplo da calculadora, vamos utilizar mais uma abordagem
+	funcional, mas dessa vez, separando algumas responsabilidades.
+	- Primeiro, crie um objeto chamado `operation` que terá as propriedades:
+	'+', '-', '*', '/' e '%'.
+	- Cada propriedade vai receber uma função (logo, elas serão métodos), e essa
+	função receberá dois parâmetros e retornará a operação referente à sua
+	propriedade, usando os valores passados por parâmetro.
+	*/
+	var operation = { 
+		'+': function ( number1, number2 ) { return number1 + number2; }, 
+		'-': function ( number1, number2 ) { return number1 - number2; },
+		'*': function ( number1, number2 ) { return number1 * number2; },
+		'/': function ( number1, number2 ) { return number1 / number2; },
+		'%': function ( number1, number2 ) { return number1 % number2; }
+	};
+	/*
+	Crie uma função chamada `isOperatorValid`, que receberá um operador por
+	parâmetro.
+	- Essa função será responsável por verificar se o operador passado por
+	parâmetro a ela é válido, ou seja, se ele é igual a '+', '-', '*', '/' ou
+	'%'.
+	- Se for igual a qualquer um desses, ela deverá retornar "true".
+	Caso contrário, "false".
+	- O desafio é fazer o retorno sem usar "if" ou "switch".
+	*/
+	function isOperatorValid( operator ){
+		return true ? operator === '+' || operator === '-' || operator === '*' || operator === '/' || operator === '%' : false;
```

Comment:
-1
Esse ternário tá meio errado, haha! Ele nunca vai entrar no último false, porque você está verificando `true` no início dele. É como se você perguntasse: `true é true?` :)

Mas você deve ter visto que, ainda assim, ele está fazendo o retorno de forma correta. Tente ajustar essa verificação, mas sem usar ternário, nem if, nem switch :)